app/Jobs/Warnings/WarningFetcher.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+declare(strict_types=1);
+
+namespace VOSTPT\Jobs\Warnings;
+
+use Carbon\Carbon;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\DB;
+use Psr\Log\LoggerInterface;
+use Ramsey\Uuid\Uuid;
+use VOSTPT\Models\IpmaWarning;
+use VOSTPT\ServiceClients\Contracts\IpmaServiceClient;
+
+class WarningFetcher implements ShouldQueue
+{
+    use Dispatchable;
+    use Queueable;
+    use SerializesModels;
+
+    const ALLOWED_AWARENESS_LEVELS = [
+        'yellow',
+        'red',
+    ];
+
+    /**
+     * @var IpmaServiceClient
+     */
+    private $serviceClient;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Execute the job.
+     *
+     * @param IpmaServiceClient        $serviceClient
+     * @param \Psr\Log\LoggerInterface $logger
+     *
+     * @return bool
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     *
+     */
+    public function handle(IpmaServiceClient $serviceClient, LoggerInterface $logger): bool
+    {
+        $this->serviceClient = $serviceClient;
+        $this->logger        = $logger;
+
+        $this->fetchIpmaWarnings();
+
+        return true;
+    }
+
+    /**
+     * Fetch Ipma warnings.
+     *
+     * @return bool
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     *
+     */
+    private function fetchIpmaWarnings(): bool
+    {
+        $this->logger->info('Fetching Ipma warnings...');
+
+        $response = $this->serviceClient->getWarnings();
+
+        foreach ($response['data'] as $data) {
+            $awarenessLevelId   = \mb_strtolower($data['awarenessLevelID']);
+            $awarenessIsAllowed = \in_array($awarenessLevelId, self::ALLOWED_AWARENESS_LEVELS, true);
+            if (empty($awarenessLevelId) || ! $awarenessIsAllowed) {
+                continue;
+            }
+
+            DB::transaction(function () use ($data, $awarenessLevelId) {
```

Comment:
We're probably better off by just fetching and serving the data with a 30min and cache, instead of storing it. Then, we can get rid of the closer logic.